membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -261,7 +261,7 @@ class AccountController(
               .map(subs => subscriptionSelector(subscriptionName, s"the sfUser $contact", subs)),
           )
           contributionPlan <- SimpleEitherT.fromEither(subscription.plan match {
-            case p: SubscriptionPlan.Contributor @unchecked /* extra guard needed due to type erasure --> */ if p.product == Contribution => Right(p)
+            case p if p.product == Contribution => Right(p)
```

Comment:
before the check was being carried forward into the type system, now it's a runtime only check.